Add/controls factory

Stick-2357 opened the pull request.

Stick-2357 committed:
Add controlsFactory

Stick-2357 committed:
Add all subsystems

shadwstalkr commented on robotbase/src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -76,12 +83,19 @@ public RobotContainer() {
-    // m_gunnerControls.m_rightTrigger.whileActiveContinuous(new ShootCommand(m_shootSub, m_gunnerControls));
-    m_gunnerControls.m_leftTrigger.whileActiveContinuous(new IntakePickupBallCommand(m_intakeSub, m_gunnerControls));
-    m_gunnerControls.m_yButton.whenPressed(new IntakeToggleDirectionCommand(m_intakeSub));
+    Map<String, ClosedLoopSubsystem> subsystems = new HashMap<>();
```
I don't like the idea of stringly typed code when it's not necessary. You should just make a class that has a member variable for each subsystem and initialize them to null. Then you don't have to cast when you use them, either. But again, you're jumping through a lot of hoops just to keep two functions out of RobotContainer.

Stick-2357 replied:
I started moving the subsystems into a separate class, but I realized it was probably a bit better to pass in an array. While getting the contents of configureButtonBindings is one of the reasons for making this change, it also is helpful for testing, as it makes it so that you can comment out one line to remove all commands dependent on a certain subsystem in one line. Once we bind all of the commands to their respective controls, the method would get huge, so I think that getting in out of RobotContainer is a near necessary thing.

Stick-2357 committed:
Use array over map

Replace map passed into the two build methods to an array passed into the constructor, and then saved onto member variables

Stick-2357 committed:
Remove unrelated uncomment

Stick-2357 committed:
Add in Vision and Drive sub

Stick-2357 committed:
Remove configureDriveSub

Stick-2357 committed:
Pass subsystems in with anonymous array

Stick-2357 committed:
Add all subsystems

Stick-2357 committed:
Merge branch 'master' into add/controls-factory

Stick-2357 committed:
Update master changes

Stick-2357 committed:
comment back in configureShuffleboard

Stick-2357 committed:
Update subsystemFactory method name

Stick-2357 committed:
Change buildControls commands to take controllers as params

Stick-2357 committed:
Pass in new Controls objects

Stick-2357 committed:
Use switch over ifs
